src/System.Management.Automation/engine/Utils.cs

```diff
@@ -895,113 +895,61 @@ internal static bool IsAdministrator()
-                    // GetFileAttributes may return INVALID_FILE_ATTRIBUTES for a system file or directory because of this error.
-                    // GetFileAttributes function tries to open the file with FILE_READ_ATTRIBUTES access right but it fails if the
-                    // sharing flag for the file is set to 0x00000000.This flag prevents it from opening a file for delete, read, or
-                    // write access. For example: C:\pagefile.sys is always opened by OS with sharing flag 0x00000000.
-                    // But FindFirstFile is still able to get attributes as this api retrieves the required information using a find
-                    // handle generated with FILE_LIST_DIRECTORY access.
-                    // Fall back to FindFirstFile to check if the file actually exists.
-                    IntPtr INVALID_HANDLE_VALUE = new IntPtr(-1);
-                    NativeMethods.WIN32_FIND_DATA findData;
-                    IntPtr findHandle = NativeMethods.FindFirstFile(path, out findData);
-                    if (findHandle != INVALID_HANDLE_VALUE)
-                    {
-                        isDirectory = (findData.dwFileAttributes & NativeMethods.FileAttributes.Directory) != 0;
-                        NativeMethods.FindClose(findHandle);
-                        return true;
-                    }
-                }
-                else if (errorCode == 53)
-                {
-                    // ERROR_BAD_NETPATH - The network path was not found.
-                    Win32Exception win32Exception = new Win32Exception(errorCode);
-                    exception = new IOException(win32Exception.Message, win32Exception);
-                }
-
-                isDirectory = false;
                 return false;
             }
-
-            isDirectory = (result & ((int)NativeMethods.FileAttributes.Directory)) ==
-                ((int)NativeMethods.FileAttributes.Directory);
-
-            return true;
-#endif
         }
 
-        // This is done through P/Invoke since we pay 13% performance degradation
-        // through the CAS checks required by File.Exists and Directory.Exists
-        internal static bool NativeFileExists(string path)
+        internal static bool FileExists(string path)
         {
-            bool isDirectory;
-            Exception ioException;
-
-            bool itemExists = NativeItemExists(path, out isDirectory, out ioException);
-            if (ioException != null)
-            {
-                throw ioException;
-            }
+            bool itemExists = ItemExists(path, out bool isDirectory);
 
             return (itemExists && (!isDirectory));
         }
 
-        // This is done through P/Invoke since we pay 13% performance degradation
-        // through the CAS checks required by File.Exists and Directory.Exists
-        internal static bool NativeDirectoryExists(string path)
+        internal static bool DirectoryExists(string path)
```

Comment:
I think we should change the name of this one and `FileExists` to something that indicate it might throw exception, so it's easy to tell whether you should use them instead of `File.Exists` and `Directory.Exists` when writing code.

I suggest rename them to `CheckFileExistsAndThrowOnError` and `CheckDirectoryExistsAndThrowOnError`.

Reply:
I'll rename if we will not migrate to File.Exists/Directory.Exists in the PR.

Reply:
No, let's use a separate PR to do the migration changes. Go ahead rename this method.